### ISSUE TITLE
Staging Review finding: Links open in a new tab without warning

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#88444]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Appointments
**Product:** Appointments
**Feature:** Appointment Details Redesign

## Findings details
**VA.gov Experience Standard - issue:** [User is taken out of the primary flow without notification.](https://design.va.gov/about/experience-standards/#user-is-notified-if-an-action-will-take-them-out-of-the-primary-flow)
**VA.gov Experience Standard - category:** [Efficiency](https://design.va.gov/about/experience-standards/#category-efficiency)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @erinrwhite (IA)

### Description

Links in the application footer, the link to “Go to My VA Health” on the appointments list, and link to directions in Google Maps on appointments details page open in new tab without warning.

## Recommended action

Any links that open in a new tab should include link text at the end of the link that says “(opens in a new tab)”. That said, because users are not in a form flow where they might lose form prgress, links should open in the same tab. See [DS guidance on link behavior](https://design.va.gov/components/link/#behavior) for more information.

## References

- **Accessibility Defect Severity:** N/A
- **WCAG Success Criteria:** N/A
- **Modality:** N/A
- **Design System Component:** [Link ](https://design.va.gov/components/link)
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** [Links](https://design.va.gov/content-style-guide/links)
- **Context:** User flow

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by sterkenburgsara** [comment]
For some reason these images aren't rendering for me, but yes @outerpress what you have above sounds right to me. The most updated guidance is to avoid opening links in another tab unless they are task interruptions (such as the scheduling flow), but I defer to OCTO on whether there is value in departing from that for links that go out to the My VA Health (OH) platform. // cc @kfink24 and @kristenmcconnell @mnorthuis @DanielleThierryUSDSVA

**Comment by mnorthuis** [comment]
@erinrwhite and @sterkenburgsara are correct on the guidance - links should not open in a new tab.  The only exception, is if the user is in a form flow and a link will cause them to lose data.  I cannot see the attached images above, so I can't confirm that is the case for your links or not, but if the person is in the process of scheduling an appointment, ideally we would avoid offering links out of the experience all together, but if we have to, they can open new tabs and should have link text that indicates that.

**Comment by outerpress** [comment]
Thanks all! These links appear in both the scheduling flow and the "manage appointment" list pages, so it sounds like we'll need to treat them differently based on context. 

Also good to know that we're not treating My VA Health/OH or the Mobile feedback forms as external pages.

**Comment by outerpress** [comment]
@ldelacosta I added design ticket #88871 to handle this.

**Comment by melissa-lefevre** [comment]
![Image](https://github.com/user-attachments/assets/58c4f70e-37a4-4bb0-9660-f109d8df3b71)
![Image](https://github.com/user-attachments/assets/a77be9fb-62a1-4078-8d65-c263298d203d)

**Comment by mnorthuis** [comment]
@melissa-lefevre What is the reason for opening these links in a new tab?  Our standard is to avoid new tabs/windows unless the user is going to lose their data (such as in a form flow), which is what Erin gets to in their feedback.  New tabs create a11y issues and are a cognitive load on mobile devices for many.  Standards on VADS: https://design.va.gov/components/link/

**Comment by erinrwhite** [comment]
And to add on here - if the link does indeed need to open in a new tab, `(opens in a new tab)` should be included as part of the link text, not adjacent to the link, so that screen readers will read it out when they read the link.

**Comment by outerpress** [comment]
@mnorthuis @erinrwhite the links under Need Help appear everywhere in appointments - including in the scheduling flow. So based on the guidance it seems like we should support two states:

- In the scheduling flow, open links in a new tab.
- Outside of that, do not open in a new tab.

Thoughts?

And @erinrwhite for the directions links, are you okay with those not opening in a new tab? Adding the language came out of the recs in this ticket, but we it sounds like we should just change those so they open in the same window.

**Comment by erinrwhite** [comment]
@outerpress 

> for the directions links, are you okay with those not opening in a new tab? 

Yes! Those can open in the same tab since it's not in a form flow.

> For the links under Need Help...based on the guidance it seems like we should support two states: In the scheduling flow, open links in a new tab. Outside of that, do not open in a new tab.

This is ideal but/and I know that this is technically difficult. Many forms on VA.gov do not do this (links in the `Need help` section always open in the same tab). Because the `Need help` section is not part of the unique page content - not part of the form content, but more of a standard footer appearing in many forms - I'd say this is okay to leave as is, opening in same window. 

@mnorthuis is out of office for a good part of the next week but I think she'd back me up on these - thanks again @outerpress for the detailed followups.

**Comment by outerpress** [comment]
Thanks @erinrwhite ! So at least for now we'll go with:
* Changing directions links so they don't open in a new tab
* Leave the links alone - they will continue to open in the same tab even in the scheduling flow (this should be fine, we have a warning when users click a link that leaves the flow)

It'll be a bit before we can get this into dev so if @mnorthuis weighs in with a different path we can pivot :)

**Comment by mnorthuis** [comment]
@erinrwhite @outerpress Agree with the final direction!

@erinrwhite let's make a note to refine guidance to make these scenarios a bit more clear.

**Comment by outerpress** [comment]
Dev ticket - #96729

**Comment by outerpress** [comment]
Dev work is done, closing this CC ticket. FYI @JRRoof
